### PR TITLE
store supplier prices

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: HappyTravel.Edo.Data/Suppliers/SupplierOrder.cs [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+using HappyTravel.Money.Enums;
 
 namespace HappyTravel.Edo.Data.Suppliers
 {
     public class SupplierOrder
     {
         public int Id { get; set; }
         public Common.Enums.Suppliers Supplier { get; set; }
-        public decimal Price { get; set; }
+        public decimal PriceInUsd { get; set; }
```
Comment (mingazhev) [inline HappyTravel.Edo.Data/Suppliers/SupplierOrder.cs:11]: Please store the original price in original currency too, as discussed

Comment (DFSko) [reply HappyTravel.Edo.Data/Suppliers/SupplierOrder.cs]: Fixed

File: Api/Models/Accommodations/BookingAvailabilityInfo.cs [hunk 2]
```diff
@@ -66,7 +69,8 @@ namespace HappyTravel.Edo.Api.Models.Accommodations
         public int NumberOfNights { get; }
         public Suppliers Supplier { get; }
         public List<AppliedMarkup> AppliedMarkups { get; }
-        public decimal SupplierPrice { get; }
+        public MoneyAmount ConvertedPrice { get; }
+        public MoneyAmount SupplierPrice { get; }
```
Comment (mingazhev) [inline Api/Models/Accommodations/BookingAvailabilityInfo.cs:73]: Both of these prices are supplier prices.
Maybe renaming to something like
ConvertedSupplierPrice
OriginalSupplierPrice
?

**Review by kirillta — APPROVED** [review]

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs [hunk 3]
```diff
@@ -41,9 +42,14 @@ public class BookingEvaluationService : IBookingEvaluationService
             var (_, isFailure, result, error) = await GetSelectedRoomSet(searchId, resultId, roomContractSetId);
             if (isFailure)
                 return ProblemDetailsBuilder.Fail<RoomContractSetAvailability?>(error);
+
+            var evaluationOnConnectorResult = await EvaluateOnConnector(result);
+            if (evaluationOnConnectorResult.IsFailure)
+                return Result.Failure<RoomContractSetAvailability?, ProblemDetails>(evaluationOnConnectorResult.Error);
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs:48]: Please use problem details builder there too. We don't need to return connector error codes there

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs [hunk 4]
```diff
@@ -41,9 +42,14 @@ public class BookingEvaluationService : IBookingEvaluationService
             var (_, isFailure, result, error) = await GetSelectedRoomSet(searchId, resultId, roomContractSetId);
             if (isFailure)
                 return ProblemDetailsBuilder.Fail<RoomContractSetAvailability?>(error);
+
+            var evaluationOnConnectorResult = await EvaluateOnConnector(result);
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs:46]: connectorEvaluationResult maybe?

File: Api/Models/Accommodations/BookingAvailabilityInfo.cs (hunk 2 above)
Comment (DFSko) [reply Api/Models/Accommodations/BookingAvailabilityInfo.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs (hunk 4 above)
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationService.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**